sstables/mx/reader.cc

```diff
@@ -441,53 +441,49 @@ class data_consume_rows_context_m : public data_consumer::continuous_data_consum
             _is_first_unfiltered = false;
             _null_component_occured = false;
             setup_ck(_column_translation.clustering_column_value_fix_legths());
-        ck_block_label:
-            if (no_more_ck_blocks()) {
-                if (_reading_range_tombstone_ck) {
-                    _reading_range_tombstone_ck = false;
-                    goto range_tombstone_body_label;
+            while (!no_more_ck_blocks()) {
```

Comment:
It's probably not your fault, but for some reason I was not able to follow this specific patch.